presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformCorrelatedSingleRowSubqueryToProject.java

```diff
@@ -95,14 +97,27 @@ public PlanNode visitLateralJoin(LateralJoinNode lateral, RewriteContext<PlanNod
             if (subqueryProjections.size() == 0) {
                 return rewrittenLateral.getInput();
             }
-            else if (subqueryProjections.size() == 1) {
-                Assignments assignments = Assignments.builder()
-                        .putIdentities(rewrittenLateral.getInput().getOutputSymbols())
-                        .putAll(subqueryProjections.get(0).getAssignments())
-                        .build();
-                return projectNode(rewrittenLateral.getInput(), assignments);
+
+            List<ProjectNode> reverseSubqueryProjection = new ArrayList<>(subqueryProjections);
+
+            Collections.reverse(reverseSubqueryProjection);
+
+            Assignments assignments = Assignments.builder().putIdentities(rewrittenLateral.getInput().getOutputSymbols())
+                    .putAll(reverseSubqueryProjection.get(0).getAssignments())
+                    .build();
+
+            ProjectNode previousNode = projectNode(rewrittenLateral.getInput(), assignments);
+
+            for (int i = 1; i < reverseSubqueryProjection.size(); i++) {
+
+                    Assignments innerAssignments = Assignments.builder().putIdentities(previousNode.getOutputSymbols())
+                            .putAll(reverseSubqueryProjection.get(i).getAssignments())
```

Comment:
You need to make sure that: `reverseSubqueryProjection.get(i).getAssignments()` does not use `reverseSubqueryProjection.get(i-1).getOutputSymbols()`